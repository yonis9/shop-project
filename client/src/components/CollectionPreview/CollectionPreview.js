import React from 'react';
import { withRouter } from 'react-router-dom';

import CollectionItem from '../CollectionItem/CollectionItem';

import {
    CollectionPreviewContainer,
    TitleContainer,
    PreviewContainer
} from './CollectionPreview.styles';

export const CollectionPreview = ({ title, items, routeName, history, match}) => (
    <CollectionPreviewContainer>
        <TitleContainer onClick={() => history.push(`${match.path}/${routeName}`)}>
            {title.toUpperCase()}
        </TitleContainer>
        <PreviewContainer>
            {
                items.filter((item, i) => i < 4)
                    .map((item) => (
                        <CollectionItem key={item.id} item={item}/>
                    ))
            }
        </PreviewContainer>
    </CollectionPreviewContainer>
)

export default withRouter(CollectionPreview);