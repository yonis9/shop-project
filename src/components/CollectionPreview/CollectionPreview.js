import React from 'react';
import './CollectionPreview.scss';
import CollectionItem from '../CollectionItem/CollectionItem';

const CollectionPreview = ({ title, items}) => (
    <div className='collection-preview'>
        <h1>{title.toUpperCase()}</h1>
        <div className='preview'>
            {
                items.filter((item, i) => i < 4)
                    .map(({ id, ...otherItemProps}) => (
                        <CollectionItem key={id} {...otherItemProps}/>
                    ))
            }
        </div>
    </div>
)

export default CollectionPreview;