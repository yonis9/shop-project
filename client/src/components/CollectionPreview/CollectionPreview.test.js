import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from '../CollectionPreview/CollectionPreview';

// title, items, routeName, history, match

describe('CollectionPreview component', () => {
    let wrapper;
    let mockHistory;
    let mockMatch;
    let mockRouteName = 'men'

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        };

        mockMatch = {
            path: 'shop'
        };

        const mockProps = {
            title: 'men',
            items: [],
            routeName: mockRouteName,
            match: mockMatch,
            history: mockHistory
        };

        wrapper = shallow(<CollectionPreview {...mockProps} />);
    });

    it('should renderd CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should called history.push with the right string when TitleContainer clicked', () => {
        const expectedArgument = `${mockMatch.path}/${mockRouteName}`;
        wrapper.find('TitleContainer').simulate('click');

        expect(mockHistory.push).toHaveBeenCalledWith(expectedArgument);
    });
});