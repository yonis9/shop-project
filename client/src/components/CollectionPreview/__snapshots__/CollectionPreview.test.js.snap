// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should renderd CollectionPreview component 1`] = `ShallowWrapper {}`;
